src/model/channel/message.rs

```diff
@@ -500,53 +427,17 @@ impl Message {
     ///
     /// # Errors
     ///
-    /// If the `cache` is enabled, returns a [`ModelError::InvalidPermissions`] if the current user
-    /// does not have the required [permissions].
+    /// Returns [`Error::Http`] if the current user lacks permission or if invalid data is given.
     ///
     /// [Add Reactions]: Permissions::ADD_REACTIONS
     /// [permissions]: crate::model::permissions
-    pub async fn react(
-        &self,
-        cache_http: impl CacheHttp,
-        reaction_type: impl Into<ReactionType>,
-    ) -> Result<Reaction> {
-        self._react(cache_http, reaction_type.into()).await
+    pub async fn react(&self, http: &Http, reaction_type: impl Into<ReactionType>) -> Result<()> {
+        self._react(http, reaction_type.into()).await
     }
 
-    async fn _react(
-        &self,
-        cache_http: impl CacheHttp,
-        reaction_type: ReactionType,
-    ) -> Result<Reaction> {
-        #[cfg_attr(not(feature = "cache"), allow(unused_mut))]
-        let mut user_id = None;
-
-        #[cfg(feature = "cache")]
-        {
-            if let Some(cache) = cache_http.cache() {
-                if let Some(guild_id) = self.guild_id {
-                    utils::user_has_perms_cache(
-                        cache,
-                        guild_id,
-                        self.channel_id,
-                        Permissions::ADD_REACTIONS,
-                    )?;
-                }
-
-                user_id = Some(cache.current_user().id);
-            }
-        }
-
-        cache_http.http().create_reaction(self.channel_id, self.id, &reaction_type).await?;
-
-        Ok(Reaction {
-            channel_id: self.channel_id,
-            emoji: reaction_type,
-            message_id: self.id,
-            user_id,
-            guild_id: self.guild_id,
-            member: self.member.as_deref().map(|member| member.clone().into()),
-        })
+    async fn _react(&self, http: &Http, reaction_type: ReactionType) -> Result<()> {
+        http.create_reaction(self.channel_id, self.id, &reaction_type).await?;
+        Ok(())
```

Comment:
This method can be gotten rid of now as there is no benefit to the de-monomorphisation.

Reply:
Resolved.